### PR TITLE
Test pr19

### PR DESCRIPTION
**Pull request by orliac**
This PR mainly implements natural weighting and the possibility to consider negative eigenvalues (off by default).
Principally tested for LSQ filter (and considering negative eigenvalues) on various datasets with comparison to WSClean & CASA with OSKAR simulated data and real LOFAR and MWA data (c.f. BIPP paper). For remaining filters I only tested consistency over {SS, NUFFT} x {CPU, GPU} solutions. In all cases sensitivity corrections were not applied.

This PR also applies WEIGHT_SPECTRUM if the column is available in the MS file.

### CODE REVIEW

**Review by AdhocMan — CHANGES_REQUESTED** [review]
We now have a CI setup for testing on Piz Daint with GPUs. Could you merge the current main branch, so we test both the new CI and these changes?

File: python/bipp/parameter_estimator.py [hunk 1]
```diff
@@ -170,12 +172,17 @@ def infer_parameters(self):
             if not np.allclose(S, 0):
                 _, D, _ = bipp.pybipp.eigh(self._ctx, S.data.shape[0], S.data, G.data)
                 # only consider positive eigenvalues, since we apply the log function for clustering
+                N = D[D <= 0.0]
                 D = D[D > 0.0]
-                idx = np.clip(np.cumsum(D) / np.sum(D), 0, 1) <= self._sigma
-                D = D[idx]
-                D_all[i, : len(D)] = D
-
+                if fne:
```
[the remark this answers is not in the document]
Comment (orliac) [reply python/bipp/parameter_estimator.py:177]: I added some inline documentation.
sigma is only considered when filtering out the negative eigenvalues (default behavior).
Also, there was a bug there as BIPP was filtering out the largest eigenvalues while it is supposed to filter the smallest eigenvalues. See https://github.com/epfl-radio-astro/bipp/pull/20/commits/53daca45397f53123c27fd65db2bee971fcf0e3f

File: src/gpu/nufft_3d3.cpp [hunk 2]
```diff
@@ -46,6 +47,8 @@ Nufft3d3<float>::Nufft3d3(int iflag, float tol, std::size_t numTrans, std::size_
                           const float* t, const float* u) {
   cufinufft_opts opts;
   cufinufftf_default_opts(3, 3, &opts);
+  opts.gpu_method = 1;
```
[the remark this answers is not in the document]
Comment (orliac) [reply src/gpu/nufft_3d3.cpp]: Suggestion committed.

File: src/standard_synthesis.cpp [hunk 3]
```diff
@@ -101,9 +101,24 @@ struct StandardSynthesisInternal {
     ctx_->logger().log_matrix(BIPP_LOG_LEVEL_DEBUG, "XYZ", nAntenna_, 3, xyz, ldxyz);
 
     const auto start = std::chrono::high_resolution_clock::now();
-
+
```
[the remark this answers is not in the document]
Comment (orliac) [reply src/standard_synthesis.cpp:104]: Suggestion committed.

File: python/bipp/frame.py [hunk 4]
```diff
@@ -53,8 +53,10 @@ def make_grids(grid_size, FoV, field_center):
             (3, grid_size, grid_size) image grid
     """
     lim = np.sin(FoV / 2)
-    grid_slice = np.linspace(-lim, lim, grid_size)
-    l_grid, m_grid = np.meshgrid(grid_slice, grid_slice)
+    offset = lim / (grid_size / 2) * 0.5
```
[the remark this answers is not in the document]
Comment (orliac) [reply python/bipp/frame.py:63]: Done in https://github.com/epfl-radio-astro/bipp/pull/20/commits/ab595b4f2884506129ed5fb2f6476266d48e76eb

**Comment by orliac** [comment]
Closing as out of date now with PR https://github.com/epfl-radio-astro/bipp/pull/28

**Pull request closed without merging**